twindb_infrastructure/providers/aws.py

```diff
@@ -25,61 +25,76 @@
 ]
 
 
-def ec2_describe_instance(instance_id):
-    cmd = [
-        "aws", "ec2", "describe-instances",
-        "--output", "json",
-        "--instance-ids", instance_id
-    ]
+class AwsError(Exception):
+    """General error in AWS"""
 
-    try:
-        log.debug("Executing: %r" % cmd)
-        aws_process = Popen(cmd, stdout=PIPE, stderr=PIPE)
-        cout, cerr = aws_process.communicate()
 
-        if aws_process.returncode != 0:
-            log.error(cerr)
-            return None
+def ec2_describe_instance(instance_id):
+    """
+    Describe EC2 instance by instance id and return an object with its
+    properties
 
-        try:
-            return json.loads(cout)
-        except ValueError as err:
-            log.error(err)
-            log.error(cerr)
-            return None
-    except OSError as err:
-        log.error(err)
-        return None
+    :param instance_id: Instance id
+    :type instance_id: str
+    :raise: AwsError
+    :return: object with instance properties
+    :rtype: dict
+    """
+    try:
+        client = boto3.client('ec2')
+        return client.describe_instances(InstanceIds=[instance_id])
+    except ClientError as err:
+        raise AwsError(err)
+    except ValueError as err:
+        raise AwsError('Internal bug in boto3: %s', err)
 
 
 def get_instance_state(instance_id):
-    response = ec2_describe_instance(instance_id)
-
+    """
+    Get state of instance by instance id
+    :param instance_id: Instance id
+    :type instance_id: str
+    :return: instance state
+    :rtype str
+    """
     try:
+        response = ec2_describe_instance(instance_id)
         state = response["Reservations"][0]["Instances"][0]["State"]["Name"]
         log.debug("Instance: %s, State: %s" % (instance_id, state))
         return state
-    except ValueError as err:
+    except AwsError as err:
```

Comment:
I think it will be better if this and all other functions raised `AwsError`.
That way we won't need to check return value in upper functions.